finetune/model.py

```diff
@@ -83,8 +86,32 @@ def masked_language_model_op(X, M, mlm_weights, mlm_ids, mlm_positions, params,
         hidden=featurizer_state["sequence_features"],
         train=(mode == tf.estimator.ModeKeys.TRAIN)
     )
-    return language_model_state
+    lm_logits = language_model_state['logits']
 
+    lm_logit_mask = np.zeros(
+        [1, lm_logits.get_shape().as_list()[-1]], dtype=np.float32
+    )
+    lm_logit_mask[:, encoder.vocab_size :] = -np.inf
+
+    if "use_extra_toks" in params and not params.use_extra_toks:
```

Comment:
Since this is a block of shared code with `language_model` I would factor this out before merge.

Reply:
refactored